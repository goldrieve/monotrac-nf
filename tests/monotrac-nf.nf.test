nextflow_workflow {

  name "Test Workflow monotrac"
  script "main.nf"
  workflow "monotrac"

  test("Should integrate all modules") {

    setup {
      def testSamplesheet = file("${outputDir}/test_samplesheet.csv")
      testSamplesheet.text = "sample,read\nSVP_test,${projectDir}/tests/data/SVP.fq.gz"
    }

    when {
      params {
        samplesheet = "${outputDir}/test_samplesheet.csv"
        outdir = "${outputDir}"
      }
    }

    then {
      assert workflow.success
      assert snapshot(workflow.out).match()
    }
  }
}