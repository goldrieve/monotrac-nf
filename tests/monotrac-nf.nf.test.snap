{
    "Should integrate all modules": {
        "content": [
            {
                "0": [
                    "combined_prediction.csv:md5,9c74c14af2ac0d9a9bb724c962b7289a"
                ],
                "FINAL": [
                    "combined_prediction.csv:md5,9c74c14af2ac0d9a9bb724c962b7289a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T15:23:25.409205959"
    }
}